### ISSUE TITLE
Not able to set TTL on NS and SOA record sets in Route53

### ISSUE THREAD

**Issue by drankard** [terraform-providers/terraform-provider-aws#8778]
### Backgroud
Having the following account structure:
--master_account
---- app1_account
---- app2_account
---- app3_account

I will need the same domain structure, so that each sub accounts can manage a `aws_route53_zone` locally, an example could be that app1 developers need to be able to manage and add the record sets needed for app1 without having to make changes in the master_account

### Problem
When creating the `aws_route53_zone` in one of the sub accounts NS and SOA record sets are generated automatically, I need to be able to change TTL on these record sets.
But found no way of doing this.

If I add an extra `aws_route53_record` for the NS records, terraform will fail complaining that the record set already exists. and i cannot add any NS or SOA parameters to the aws_route53_zone.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.13
provider.aws v2.10.0

### Affected Resource(s)

* aws_route53_zone
* aws_route53_record

### Terraform Configuration Files

```hcl
resource "aws_route53_zone" "zone" {
  name = "test.example.com"
}

resource "aws_route53_record" "ns" {
  zone_id = "${aws_route53_zone.zone.zone_id}"
  name    = "test.example.com"
  type    = "NS"
  ttl     = "30"

  records = [
    "${aws_route53_zone.zone.name_servers.0}",
    "${aws_route53_zone.zone.name_servers.1}",
    "${aws_route53_zone.zone.name_servers.2}",
    "${aws_route53_zone.zone.name_servers.3}",
  ]
}
```

### Output
```
* aws_route53_record.ns: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='app1.example.com.', type='NS'] but it already exists]
	status code: 400, request id: XXXXXX
```

**Comment by bflad** [comment]
Hi @drankard 👋 Thanks for writing this in and apologies if the documentation was not clear in this regard.

---

If you used the Terraform AWS Provider prior to version 2.0.0, this section will provide some brief insight into default behavior changes that occurred with that major release. If you have only used the provider past 2.0.0, feel free to skip this quick historical note.

Prior to version 2.0.0, the Terraform resource would automatically perform an `UPSERT` operation within Route 53 when creating a Terraform `aws_route53_record` resource instead of a `CREATE` operation. This behavior was problematic for many practitioners in the older versions of the Terraform AWS Provider and the [default behavior of the resource was changed in version 2.0.0](https://www.terraform.io/docs/providers/aws/guides/version-2-upgrade.html#resource-aws_route53_record).

---

The majority of Terraform resources take the stance that creating the resource in Terraform should generally produce an error if the infrastructure component already exists. This is to help prevent conflicting management of infrastructure by ensuring that practitioners opt into adopting existing infrastructure into their configurations/state via `terraform import`. The [import section of the `aws_route53_record` resource](https://www.terraform.io/docs/providers/aws/r/route53_record.html#import) tries to show how this can be accomplished in general. With this case, a command like the following should help get you started where the zone identifier (Z123...) should be substituted to match the actual Route 53 Zone ID:

```console
$ terraform import aws_route53_record.ns Z1234567890_test.example.com_NS
```

Understandably though, this is particularly unhelpful in the `NS` record case since these are automatically created by Route 53 with the creation of the zone. With the `aws_route53_record` resource we do allow the behavior of overwriting an existing Route 53 Record via the [`allow_overwrite` argument](https://www.terraform.io/docs/providers/aws/r/route53_record.html#allow_overwrite), which ensures you can redefine the `NS` record without needing to import the record for each new zone. An example of this with the above configuration would be:

```hcl
resource "aws_route53_zone" "zone" {
  name = "test.example.com"
}

resource "aws_route53_record" "ns" {
  allow_overwrite = true
  zone_id         = "${aws_route53_zone.zone.zone_id}"
  name            = "test.example.com"
  type            = "NS"
  ttl             = "30"

  records = [
    "${aws_route53_zone.zone.name_servers.0}",
    "${aws_route53_zone.zone.name_servers.1}",
    "${aws_route53_zone.zone.name_servers.2}",
    "${aws_route53_zone.zone.name_servers.3}",
  ]
}
```

Which will allow Terraform, even with "creating" the resource, to actually perform an upsert so the `terraform import` is unnecessary. We typically will recommend that the `allow_overwrite` argument only be used where necessary (such as this case), since it can cause unexpected behaviors for Terraform practitioners and we do not intend to include similar arguments/behaviors for other Terraform resources.

Hope this helps! 👍 I'm going to keep this issue open to note that we should add an example of this NS record configuration in the resource documentation since this has come up previously as well.

**Comment by bflad** [comment]
Documentation has been merged and will publish when version 2.14.0 of the Terraform AWS Provider is released later this week. 👍
